client/src/test/java/io/spine/client/QueryBuilderSpec.kt

```diff
@@ -0,0 +1,598 @@
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.client
+
+import com.google.common.testing.NullPointerTester
+import com.google.common.truth.Subject
+import com.google.common.truth.Truth.assertThat
+import com.google.protobuf.Int32Value
+import com.google.protobuf.Message
+import com.google.protobuf.Timestamp
+import com.google.protobuf.util.Durations
+import com.google.protobuf.util.Timestamps.subtract
+import io.spine.base.EntityState
+import io.spine.base.Identifier.newUuid
+import io.spine.base.Time
+import io.spine.client.CompositeFilter.CompositeOperator.ALL
+import io.spine.client.CompositeFilter.CompositeOperator.EITHER
+import io.spine.client.Filters.all
+import io.spine.client.Filters.either
+import io.spine.client.Filters.eq
+import io.spine.client.OrderBy.Direction.ASCENDING
+import io.spine.client.OrderBy.Direction.DESCENDING
+import io.spine.client.OrderBy.Direction.OD_UNKNOWN
+import io.spine.client.OrderBy.Direction.UNRECOGNIZED
+import io.spine.client.given.TestEntities.randomId
+import io.spine.protobuf.AnyPacker
+import io.spine.protobuf.TypeConverter.toObject
+import io.spine.test.client.TestEntity
+import io.spine.test.client.TestEntityId
+import io.spine.testing.DisplayNames
+import io.spine.testing.core.given.GivenUserId
+import io.spine.time.ZoneIds
+import io.spine.type.TypeUrl
+import java.util.*
+import java.util.stream.Collectors.toList
+import kotlin.Any
+import kotlin.IllegalArgumentException
+import kotlin.Int
+import kotlin.RuntimeException
+import kotlin.String
+import kotlin.Unit
+import kotlin.apply
+import kotlin.arrayOf
+import kotlin.with
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import com.google.protobuf.Any as AnyProto
+
+@DisplayName("`QueryBuilder` should")
+internal class QueryBuilderSpec {
```

Comment:
Is there a reason for us to use `@DisplayName` instead of the class name is backticks?

I'm not against using `..Spec` in Java tests, but Kotlin gives us, IMHO, a better option.

Reply:
@armiol, @dmdashenkov, your thoughts?

Reply:
Yes, that makes sense. We'll need to include this explanation into our (future?) Kotlin codestyle guidelines.

Reply:
https://github.com/SpineEventEngine/documentation/issues/28